libvisual/CMakeLists.txt

```diff
@@ -185,14 +193,29 @@ ENDIF()
 
 # Git revision
 FIND_PACKAGE(Git)
-CONFIGURE_FILE(
-  "${PROJECT_SOURCE_DIR}/update-version.sh.in"
-  "${PROJECT_BINARY_DIR}/update-version.sh"
-  @ONLY
-)
-ADD_CUSTOM_TARGET(update-version ALL
-  COMMAND /bin/sh ${PROJECT_BINARY_DIR}/update-version.sh
-)
+
+EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} describe --always --dirty
+  OUTPUT_VARIABLE LV_REVISION OUTPUT_STRIP_TRAILING_WHITESPACE)
+IF(NOT LV_REVISION)
+  SET(LV_REVISION "unknown")
+ENDIF()
+
+IF(EXISTS ${BINARY_DIR}/version.h)
+  FILE(STRINGS ${BINARY_DIR}/version.h OLD_REVISION REGEX "#define LV_REVISION")
+  STRING(REGEX MATCH \".*\" OLD_REVISION ${OLD_REVISION})
+  STRING(REGEX MATCHALL "[^\"][0-9A-z]+" OLD_REVISION ${OLD_REVISION})
```

Comment:
Capital `A` and lowercase `z` in `A-z` looks unintended and did not work in my tests.
Also, the use of `MATCHALL` seems to be a problem with Git output like `fa8021a6-dirty`. What works for me:
```
STRING(REGEX MATCH "[^\"][0-9a-z]+" OLD_REVISION ${OLD_REVISION})
                 ^^            ^
```
I would suggest we apply the same thing to `LV_REVISION` as well so that the `-dirty` in `fa8021a6-dirty` does not trigger re-generation every time.

Reply:
Should we not use `--dirty` to append `-dirty` to the revision?

Reply:
I'm not sure. We could also make our regex allow `-` and then the `dirty` would make it's way through on both sides.